pkg/datapath/linux/ipsec/xfrm_collector.go

```diff
@@ -101,5 +124,19 @@ func (x *xfrmCollector) Collect(ch chan<- prometheus.Metric) {
 	ch <- prometheus.MustNewConstMetric(x.xfrmErrorDesc, prometheus.GaugeValue, float64(stats.XfrmOutPolDead), labelErrorTypeOutbound, labelErrorPolicyDead)
 	ch <- prometheus.MustNewConstMetric(x.xfrmErrorDesc, prometheus.GaugeValue, float64(stats.XfrmOutPolError), labelErrorTypeOutbound, labelErrorPolicy)
 	ch <- prometheus.MustNewConstMetric(x.xfrmErrorDesc, prometheus.GaugeValue, float64(stats.XfrmOutStateInvalid), labelErrorTypeOutbound, labelErrorStateInvalid)
+}
 
+func (x *xfrmCollector) collectConfigStats(ch chan<- prometheus.Metric) {
+	states, err := x.xfrmStateFunc()
+	if err != nil {
+		log.WithError(err).Error("Error while getting XFRM states")
```

Comment:
nit: would it better to return the error here and log at the collect level with some context of what the outcome of this error is.